### DISCUSSION TITLE
How to migrate project compile with latest axmol(CC prefix was removed from code file name)

### DISCUSSION THREAD

**Discussion by halx99** [axmolengine/axmol#1231] [Announcements]
## Migrate guide

If you are include engine headers not only `cocos2d.h`/`axmol.h`, then you should read this guide for quick fix your header includes of your project sources.

1. clone https://github.com/axmolengine/axmol-migrate
2. Goto folder `axmol-migrate` and run `pwsh ./build.ps1 -xb "'--config', 'Release'"`
4. use `axmol-migrate` to migrate your project sources folder
    `./build_x64/bin/Release/axmol-migrate source --fuzzy --source-dir path/to/your/project/sources/folder` 

## Notes

- The command line `axmol-migrate.exe` can fix 99% or 100% for some project (i.e. [OpenGD](https://github.com/axmolengine/axmol-migrate/actions/runs/5485636126/jobs/9994685016): ) of your includes, after execute it, you can manually fix compile error of you project. 
- The `axmol-migrate` can compile on windows, macos, linux

refer to: #1230

**Comment by paulocoutinhox** [comment]
Can you create a python tool instead of .exe that only run on Windows?

**Reply by halx99** [reply]
the sources-migrate can compile on windows, linux, macos

**Comment by cigumo** [comment]
I have a question about CC_* macros. I see there are some compatibility entries in axmol/cocos2d.h. 
What's the idea? To migrate CC_* macros in the project to AX_*? Or should we make cocos2d.h a complete list? 

Some examples:
CC_RADIANS_TO_DEGREES
CC_CALLBACK_3
NS_CC_BEGIN and NS_CC_END
CC_SAFE_DELETE_ARRAY

Thanks.
